src/middlewared/middlewared/plugins/iscsi_/extents.py

```diff
@@ -80,9 +81,18 @@ class Config:
     @private
     async def sharing_task_determine_locked(self, data):
         """Determine if this extent is in a locked path"""
+        path = await self.get_path_field(data)
+        if data['type'] == 'FILE' and path.startswith('/mnt'):
+            # Sanitize the path, remove non ds components that are invalid names
+            path_ = path.removeprefix('/mnt/')
+            while path_:
+                if validate_dataset_name(path_):
```

Comment:
I don't see a reason for this `validate_dataset_name` call and the extra "path.startswith("/mnt")` logic. We already remove the "/mnt/" prefix in the method itself so it seems like this is adding too much extra unnecessary logic. I pictured something like this:
```
if data['type'] == 'FILE':
    path = os.path.dirname(path)

return await self.middleware.call('pool.dataset.path_in_locked_dataset', path)

Reply:
Can't rely on `statfs` because the filesystem for which the path exists might be locked (which means unmounted) and so the path won't be available.